### PR TITLE
[Issue 11581][broker] feat:pass the executor to RateLimiter in ResourceGroupPublishLimiter

### LINKED ISSUE #11581: pass the executor to RateLimiter in  ResourceGroupPublishLimiter
**Is your enhancement request related to a problem? Please describe.**
A clear and concise description of what the enhancement is.

pass the executor to RateLimiter in  ResourceGroupPublishLimiter

**Describe the solution you'd like**

it is necessary 

**Describe alternatives you've considered**

the RateLimiter should use the executor of ResourceGroupPublishLimiter

**Additional context**

### PR DESCRIPTION
**Pull request by leizhiyuan**
### background

Fixes #11581 

### Motivation

we need to pass the executor to RateLimiter in ResourceGroupPublishLimiter

### Modifications

 pass the executor to RateLimiter in ResourceGroupPublishLimiter

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema:(no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options:(no)
  - Anything that affects deployment: (no)

#### For contributor

For this PR, do we need to update docs?

no, there is no change for users.

### CODE REVIEW

File: pulsar-common/src/main/java/org/apache/pulsar/common/util/RateLimiter.java [hunk 1]
```diff
@@ -92,6 +92,11 @@ public RateLimiter(final ScheduledExecutorService service, final long permits, f
         this(service, permits, rateTime, timeUnit, permitUpdater, false);
     }
 
+    public RateLimiter(final ScheduledExecutorService service, final long permits, final long rateTime,
```
Comment (lhotari) [inline pulsar-common/src/main/java/org/apache/pulsar/common/util/RateLimiter.java:95]: RateLimiter already contains a lot of different constructors. The alternative is to to use the builder generated with the Lombok Builder annotation. Example of using this is in https://github.com/apache/pulsar/blob/ded806fd52f6e2f182fa02052cbd82c2a6755098/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PrecisPublishLimiter.java#L101-L117

Comment (leizhiyuan) [reply pulsar-common/src/main/java/org/apache/pulsar/common/util/RateLimiter.java:95]: ok,I will fix this.

**Comment by Anonymitaet** [comment]
Thanks for providing doc-related info!

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java [hunk 2]
```diff
@@ -95,14 +95,25 @@ public void update(ResourceGroup resourceGroup) {
...
                 if (this.publishMaxMessageRate > 0) {
-                    // TODO: pass the executor
-                    publishRateLimiterOnMessage =
-                        new RateLimiter(publishMaxMessageRate, 1, TimeUnit.SECONDS, this::apply);
+                    publishRateLimiterOnMessage = RateLimiter.builder()
+                            .scheduledExecutorService(scheduledExecutorService)
+                            .permits(publishMaxMessageRate)
+                            .rateTime(1L)
+                            .timeUnit(TimeUnit.SECONDS)
+                            .isDispatchOrPrecisePublishRateLimiter(false)
```
Comment (lhotari) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java:103]: the default value is false, so there's no need to set it

**Comment by leizhiyuan** [comment]
image

the failed tase case  runs ok in local machine

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java (hunk 2 above)
Comment (leizhiyuan) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java]: done

**Comment by lhotari** [comment]
> the failed tase case runs ok in local machine

@leizhiyuan There are quite a few flaky tests. I assume you faced #11096 .

When you suspect a flaky test, please search in the GitHub issue with the short name of the test class for possible reported flaky tests. 

It is possible to retry a failed CI build by adding a comment "/pulsarbot rerun-failure-checks" to the PR.

* Before retrying a failed build, please check the failure in the logs. Check the issue tracker whether the test failure has been reported as a flaky test. If not, please report it.
* If it has been reported, please add a comment about facing a flaky issue in a specific build (provide link to log line in GitHub Actions log).

Reporting a new Flaky issue can be done by choosing "Flaky test" on the new
issue page https://github.com/apache/pulsar/issues/new/choose .
Searching flaky issues:
https://github.com/apache/pulsar/issues?q=is%3Aissue+is%3Aopen+flaky+sort%3Aupdated-desc
(enter "flaky" or the test method name as the search query)

**Review by lhotari — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Review by 315157973 — APPROVED** [review]

**Comment by bharanic-dev** [comment]
@leizhiyuan Thank you for taking care of this. Passing the Executor was one of the tasks in my TODO list (which I why I left the comment in the code). Glad you were able to get to it before I could.

Just to clarify though, I don't think this addresses any flaky tests. The flaky tests reported in:
https://github.com/apache/pulsar/issues/11096
https://github.com/apache/pulsar/pull/11601

are because of a code issue, where the zookeeper watch events were getting fixed. The race condition was reported in #11157 and fixed in #11198

**Comment by bharanic-dev** [comment]
LGTM

**Review by sijie — APPROVED** [review]

**Pull request merged**